lib/reply.js

```diff
@@ -168,6 +170,11 @@ Reply.prototype.send = function (payload) {
       return this
     }
 
+    if (payload[kWebStreamReadable]) {
+      onSendHook(this, Readable.fromWeb(payload))
```

Comment:
You should not change the payload here.
`Readable.fromWeb` must provided on `onSendEnd`.

Reply:
...also, I think there are a few headers that should be preserved if it's a fetch `Response`.

Reply:
It would be another PR, but identify `Response` would be hard.
I don't know which `function` would be specific enough to identify it.

Reply:
Why on onSendEnd?

What if in an onSend hook i check for a stream like it is done in @fastify/throttle?

Reply:
It looks like a partial support of `WebStream` as we mutate the response in the middle.
I know that `object` will be serialized and `BufferLike` is already changed to `Buffer`.


But, `WebStream` should be special treated to allow better support or extendible like extends it to `Response` in the middle.
Ultimately, `WebStream` should be possible to do the same as `NodeStream` or even `Response`.
https://github.com/nodejs/node/issues/42529


```js
fastify.addHook('onSend', async (request, reply, payload) => {
  if (payload instanceof ReadableStream) {
    // we can use Response here
    return new Response(payload, {
      headers: { "Content-Type": "text/html" }
    })
  }
  return payload
})
```